plugins/nordic-hid/fu-nordic-hid-archive.c

```diff
@@ -11,15 +11,161 @@
+						gint64 manifest_ver,
+						GError **error)
+{
+	g_auto(GStrv) board_split = NULL;
+	const gchar *board_name_readout = NULL;
+
+	if (!json_object_has_member(obj, "board")) {
+		g_set_error_literal(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_INVALID_FILE,
+				    "manifest invalid as has no target board information");
+		return NULL;
+	}
+
+	board_name_readout = json_object_get_string_member(obj, "board");
+	if (manifest_ver == 0) {
+		/* for manifest "format-version" of "0", the board name is represented only
+		 * by part of the string before the "_" symbol
+		 */
+		board_split = g_strsplit(board_name_readout, "_", -1);
+		if (board_split[0] == NULL) {
+			g_set_error_literal(error,
+					    FWUPD_ERROR,
+					    FWUPD_ERROR_INVALID_FILE,
+					    "manifest invalid as has no board information");
+			return NULL;
+		}
+		return g_strdup(board_split[0]);
+	} else {
+		/* duplicate string for consistent memory management */
+		return g_strdup(board_name_readout);
+	}
+}
+
+static gint
+fu_nordic_hid_archive_parse_file_get_flash_area_id(JsonObject *obj,
+						   gint64 manifest_ver,
+						   guint file_idx,
+						   const gchar *bootloader_name,
+						   guint files_cnt,
+						   GError **error)
+{
+	if (manifest_ver == 0) {
+		/* the images are expected to be listed in strict order */
+		return file_idx;
+	} else {
+		gint64 image_idx;
+		gint64 slot;
+
+		/* for MCUboot bootloader with swap, if only a single image is available,
+		 * the "image_index" and "slot" properties may be omitted
+		 */
+		if (!g_strcmp0(bootloader_name, "MCUBOOT") && files_cnt == 1) {
+			return 0;
+		} else {
+			if (!json_object_has_member(obj, "image_index")) {
+				g_set_error_literal(error,
+						    FWUPD_ERROR,
+						    FWUPD_ERROR_INVALID_FILE,
+						    "missing image_index property");
+				return -1;
+			}
+			if (!fu_strtoll(json_object_get_string_member(obj, "image_index"),
+					&image_idx,
+					G_MININT64,
+					G_MAXINT64,
+					FU_INTEGER_BASE_AUTO,
+					error)) {
+				g_prefix_error(error, "fu_strtoll failed for image_index:");
+				return -1;
+			}
+
+			if (!json_object_has_member(obj, "slot")) {
+				g_set_error_literal(error,
+						    FWUPD_ERROR,
+						    FWUPD_ERROR_INVALID_FILE,
+						    "missing slot property");
+				return -1;
+			}
+			if (!fu_strtoll(json_object_get_string_member(obj, "slot"),
+					&slot,
+					G_MININT64,
+					G_MAXINT64,
+					FU_INTEGER_BASE_AUTO,
+					error)) {
+				g_prefix_error(error, "fu_strtoll failed for slot:");
+				return -1;
+			}
+
+			if (image_idx != 0) {
+				g_set_error_literal(error,
+						    FWUPD_ERROR,
+						    FWUPD_ERROR_INVALID_FILE,
+						    "unsupported image_index property");
+				return -1;
+			}
+			if (slot != 0 && slot != 1) {
+				g_set_error_literal(error,
+						    FWUPD_ERROR,
+						    FWUPD_ERROR_INVALID_FILE,
+						    "unsupported slot property");
+				return -1;
+			}
+			return slot;
+		}
+	}
+}
```

Comment:
no return here?

Reply:
It relied completely on early returns, so return here was not needed